### DISCUSSION TITLE
is the api already running?

### DISCUSSION THREAD

**Discussion by ghost** [Valour-Software/Valour#78] [Q&A] [answered]
because we are doing the bots thing and we need something to work on, if it isnt, i have kind of atemplate for it.

**Comment by identithree** [comment]
As far as I am aware, there is a framework but not much in respect of, say, logging into an account.

**Comment by superjacobl** [comment, marked as answer]
No, the API is not working right now, as CORS blocks most requests. We are planning on adding better api routes for bots. Expect basic API after pre-alpha.

**Reply by ghost** [reply]
is there at least a template for what it will look like? So we can start to work on it.

**Reply by Coca162** [reply]
It will probably just be what it is now just with different routes

**Reply by ghost** [reply]
so where can i find it?

**Reply by superjacobl** [reply]
Spike has said something about not having a separate bot api, but as of right now, we have 0 info.

**Reply by Coca162** [reply]
In the client itself? It's basically has an API wrapper (and Spike said it will become one for bots in the future for C#)

**Reply by ghost** [reply]
honk said that there would be an api, but the bots wrapper, the api itself would be complety handle by the vapp project, but is there a file (something like "apiwrapper.cs") to keep track of and build an wrapper over it?

**Reply by Coca162** [reply]
I don't know what you mean by that
